classes/pointmarking_form.php

```diff
@@ -0,0 +1,139 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Form to mark an observation point
+ *
+ * @package   mod_observation
+ * @copyright  2021 Endurer Solutions Team
+ * @author Matthew Hilton <mj.hilton@outlook.com>
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace mod_observation;
+
+defined('MOODLE_INTERNAL') || die;
+require_once($CFG->libdir.'/formslib.php');
+
+/**
+ * Creates a moodle_form to mark an observation point.
+ *
+ * @package   mod_observation
+ * @copyright  2021 Endurer Solutions Team
+ * @author Matthew Hilton <mj.hilton@outlook.com>
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class pointmarking_form extends \moodleform {
+    /**
+     * Defines the point marking form
+     */
+    public function definition() {
+        $mform = $this->_form;
+
+        $prefill = $this->_customdata;
+
+        $mform->addElement('header', 'obpointheader', get_string('observationpoint', 'observation'));
+
+        // Title.
+        $mform->addElement('static', 'title', get_string('title', 'observation'));
+
+        // Observation point information.
+        $instext = $prefill['ins'];
+        $insformat = $prefill['ins_f'];
+
+        $mform->addElement('static', 'instructions', get_string('gradinginstructions', 'observation'),
+            format_text($instext, $insformat));
+
+        switch($prefill['res_type']) {
+            // Text input type.
+            case 0:
```

Comment:
Im fine with the switch staying to support additional types in future, but the case should likely become a class constant:

`const INPUT_TEXT = 0;`